components/script/dom/performance.rs

```diff
@@ -96,10 +96,10 @@ impl PerformanceEntryList {
         entry_type: Option<DOMString>,
     ) {
         self.entries.retain(|e| {
-            name.as_ref().map_or(true, |name_| *e.name() == *name_) &&
+            name.as_ref().map_or(true, |name_| *e.name() != *name_) ||
```

Comment:
I think we either need to use map_or(false, ...) with ||, or map_or(true, ...) with &&. The current code means that if there is no name passed but the entry type matches then we will not clear the entry even though we should. Similarly, if there is no entry passed but the names match, we will ignore it.

Reply:
Thank you for the review. Oh, I misunderstood the clear condition.
Fixed it.